### ISSUE TITLE
Sometimes icons do not hide when cooldown finishes

### ISSUE THREAD

**Issue by SweepyBoop** [SweepyBoop/Sweepy-Boop#192]

**Comment by SweepyBoop** [comment]
local function OnCooldownTimerFinished(self)
    StopAnimation(self:GetParent());
    NS.FinishCooldownTimer(self);
end

NS.FinishCooldownTimer = function (self)
    NS.RefreshCooldownTimer(self, true);
end

**Comment by SweepyBoop** [comment]
```
if ( start ~= math.huge ) and ( duration ~= math.huge ) then
        icon.cooldown:SetCooldown(start, duration);
        if icon.Count then
            icon.Count:SetText(stack and "#" or "");
        end
    else
```

This also needs to check against current time

**Comment by SweepyBoop** [comment]
Fixed by #193 #194 #195
